<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>tagtango â€¢ tagtango</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="tagtango">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">tagtango</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/tagtango.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>tagtango</h1>
            
      

      <div class="d-none name"><code>tagtango.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="input-data-requirements">Input data requirements<a class="anchor" aria-label="anchor" href="#input-data-requirements"></a>
</h2>
<p><code>tagtango</code> is flexible in handling various data types. To
date, it accepts input in the form of either a
<code>MultiAssayExperiment</code> R object stored as an RDS file (<a href="https://doi.org/10.1158/0008-5472.CAN-17-0344" class="external-link">Ramos et. al.,
2017</a>), a <code>SingleCellExperiment</code> R object stored as an RDS
file (<a href="https://doi.org/10.1038/s41592-019-0654-x" class="external-link">Amezquita et.
al., 2020</a>), or an R data frame stored as an RDS, CSV, or TSV
file.</p>
<div class="section level3">
<h3 id="multiassayexperiment-object">
<code>MultiAssayExperiment</code> object<a class="anchor" aria-label="anchor" href="#multiassayexperiment-object"></a>
</h3>
<p>Providing a <code>MultiAssayExperiment</code> object as input will
allow you to study multiple data modalities simultaneously. However,
there are certain criteria that needs to be met. First, the elements of
the <code>ExperimentList</code> container should be
<code>SingleCellExperiment</code> following the specifications stated in
the next section. Second, cells in all elements of the
<code>ExperimentList</code> should be the same (and have unique and
matching names). Finally, different annotations should be stored as
columns of the <code>colData</code> data frame within the object.</p>
<p>As test dataset, a preprocessed and annotated 10x dataset is provided
with the package. This is a <code>MultiAssayExperiment</code> with
Peripheral Blood Mononuclear Cells (PBMCs) from a healthy donor stained
with a few TotalSeq-B antibodies (<a href="https://doi.org/" class="external-link">10x
Genomics, 2018</a>), and is readily accessible via:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">tagtango</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span></span>
<span><span class="va">test_data</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an</span></span></code></pre>
<pre><code><span><span class="co">## ExperimentList class object of length 2:</span></span>
<span><span class="co">##  [1] RNA: SingleCellExperiment with 33538 rows and 7472 columns</span></span>
<span><span class="co">##  [2] ADT: SingleCellExperiment with 17 rows and 7472 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>Notice that the column and row names of each
<code>SingleCellExperiment</code> in the corresponding
<code>ExperimentList</code> object are not <code>NULL</code>, and that
the different annotations are stored in:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MultiAssayExperiment</span><span class="fu">::</span><span class="fu">colData</span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 7472 rows and 21 columns</span></span>
<span><span class="co">##                                    Sample            Barcode       sum</span></span>
<span><span class="co">##                               &lt;character&gt;        &lt;character&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## AAACCCAAGATTGTGA-1 /var/folders/w9/2_9f.. AAACCCAAGATTGTGA-1      6160</span></span>
<span><span class="co">## AAACCCACATCGGTTA-1 /var/folders/w9/2_9f.. AAACCCACATCGGTTA-1      6713</span></span>
<span><span class="co">## AAACCCAGTACCGCGT-1 /var/folders/w9/2_9f.. AAACCCAGTACCGCGT-1      3637</span></span>
<span><span class="co">## AAACCCAGTATCGAAA-1 /var/folders/w9/2_9f.. AAACCCAGTATCGAAA-1      1244</span></span>
<span><span class="co">## AAACCCAGTCGTCATA-1 /var/folders/w9/2_9f.. AAACCCAGTCGTCATA-1      2611</span></span>
<span><span class="co">## ...                                   ...                ...       ...</span></span>
<span><span class="co">## TTTGTTGGTTCAAGTC-1 /var/folders/w9/2_9f.. TTTGTTGGTTCAAGTC-1      5830</span></span>
<span><span class="co">## TTTGTTGGTTGCATGT-1 /var/folders/w9/2_9f.. TTTGTTGGTTGCATGT-1      4096</span></span>
<span><span class="co">## TTTGTTGGTTGCGGCT-1 /var/folders/w9/2_9f.. TTTGTTGGTTGCGGCT-1      5524</span></span>
<span><span class="co">## TTTGTTGTCGAGTGAG-1 /var/folders/w9/2_9f.. TTTGTTGTCGAGTGAG-1      4039</span></span>
<span><span class="co">## TTTGTTGTCGTTCAGA-1 /var/folders/w9/2_9f.. TTTGTTGTCGTTCAGA-1      4213</span></span>
<span><span class="co">##                     detected subsets_Mito_sum subsets_Mito_detected</span></span>
<span><span class="co">##                    &lt;integer&gt;        &lt;numeric&gt;             &lt;integer&gt;</span></span>
<span><span class="co">## AAACCCAAGATTGTGA-1      2194              523                    11</span></span>
<span><span class="co">## AAACCCACATCGGTTA-1      2093              415                    11</span></span>
<span><span class="co">## AAACCCAGTACCGCGT-1      1518              287                    11</span></span>
<span><span class="co">## AAACCCAGTATCGAAA-1       737              110                    12</span></span>
<span><span class="co">## AAACCCAGTCGTCATA-1      1240              156                    11</span></span>
<span><span class="co">## ...                      ...              ...                   ...</span></span>
<span><span class="co">## TTTGTTGGTTCAAGTC-1      2178              323                    11</span></span>
<span><span class="co">## TTTGTTGGTTGCATGT-1      1256              256                    11</span></span>
<span><span class="co">## TTTGTTGGTTGCGGCT-1      1907              574                    11</span></span>
<span><span class="co">## TTTGTTGTCGAGTGAG-1      1605              271                    10</span></span>
<span><span class="co">## TTTGTTGTCGTTCAGA-1      1549              209                    11</span></span>
<span><span class="co">##                    subsets_Mito_percent altexps_Antibody.Capture_sum</span></span>
<span><span class="co">##                               &lt;numeric&gt;                    &lt;numeric&gt;</span></span>
<span><span class="co">## AAACCCAAGATTGTGA-1              8.49026                          981</span></span>
<span><span class="co">## AAACCCACATCGGTTA-1              6.18203                         1475</span></span>
<span><span class="co">## AAACCCAGTACCGCGT-1              7.89112                         7149</span></span>
<span><span class="co">## AAACCCAGTATCGAAA-1              8.84244                         6831</span></span>
<span><span class="co">## AAACCCAGTCGTCATA-1              5.97472                         6839</span></span>
<span><span class="co">## ...                                 ...                          ...</span></span>
<span><span class="co">## TTTGTTGGTTCAAGTC-1              5.54031                         9520</span></span>
<span><span class="co">## TTTGTTGGTTGCATGT-1              6.25000                         7763</span></span>
<span><span class="co">## TTTGTTGGTTGCGGCT-1             10.39102                         1973</span></span>
<span><span class="co">## TTTGTTGTCGAGTGAG-1              6.70958                         4316</span></span>
<span><span class="co">## TTTGTTGTCGTTCAGA-1              4.96084                         5682</span></span>
<span><span class="co">##                    altexps_Antibody.Capture_detected</span></span>
<span><span class="co">##                                            &lt;integer&gt;</span></span>
<span><span class="co">## AAACCCAAGATTGTGA-1                                17</span></span>
<span><span class="co">## AAACCCACATCGGTTA-1                                17</span></span>
<span><span class="co">## AAACCCAGTACCGCGT-1                                17</span></span>
<span><span class="co">## AAACCCAGTATCGAAA-1                                17</span></span>
<span><span class="co">## AAACCCAGTCGTCATA-1                                17</span></span>
<span><span class="co">## ...                                              ...</span></span>
<span><span class="co">## TTTGTTGGTTCAAGTC-1                                17</span></span>
<span><span class="co">## TTTGTTGGTTGCATGT-1                                17</span></span>
<span><span class="co">## TTTGTTGGTTGCGGCT-1                                17</span></span>
<span><span class="co">## TTTGTTGTCGAGTGAG-1                                17</span></span>
<span><span class="co">## TTTGTTGTCGTTCAGA-1                                16</span></span>
<span><span class="co">##                    altexps_Antibody.Capture_percent     total sizeFactor</span></span>
<span><span class="co">##                                           &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;</span></span>
<span><span class="co">## AAACCCAAGATTGTGA-1                          13.7376      7141   1.301038</span></span>
<span><span class="co">## AAACCCACATCGGTTA-1                          18.0142      8188   1.417836</span></span>
<span><span class="co">## AAACCCAGTACCGCGT-1                          66.2804     10786   0.768162</span></span>
<span><span class="co">## AAACCCAGTATCGAAA-1                          84.5944      8075   0.262742</span></span>
<span><span class="co">## AAACCCAGTCGTCATA-1                          72.3704      9450   0.551463</span></span>
<span><span class="co">## ...                                             ...       ...        ...</span></span>
<span><span class="co">## TTTGTTGGTTCAAGTC-1                          62.0195     15350   1.231340</span></span>
<span><span class="co">## TTTGTTGGTTGCATGT-1                          65.4608     11859   0.865106</span></span>
<span><span class="co">## TTTGTTGGTTGCGGCT-1                          26.3172      7497   1.166710</span></span>
<span><span class="co">## TTTGTTGTCGAGTGAG-1                          51.6577      8355   0.853067</span></span>
<span><span class="co">## TTTGTTGTCGTTCAGA-1                          57.4229      9895   0.889817</span></span>
<span><span class="co">##                       Main.ADT    Fine.ADT RNA.Azimuth.L1 RNA.Azimuth.L2</span></span>
<span><span class="co">##                    &lt;character&gt; &lt;character&gt;    &lt;character&gt;    &lt;character&gt;</span></span>
<span><span class="co">## AAACCCAAGATTGTGA-1   cluster 2   cluster 9           Mono      CD14 Mono</span></span>
<span><span class="co">## AAACCCACATCGGTTA-1   cluster 2   cluster 3           Mono      CD14 Mono</span></span>
<span><span class="co">## AAACCCAGTACCGCGT-1   cluster 2   cluster 3           Mono      CD14 Mono</span></span>
<span><span class="co">## AAACCCAGTATCGAAA-1   cluster 5   cluster 4             NK             NK</span></span>
<span><span class="co">## AAACCCAGTCGTCATA-1   cluster 5   cluster 4             NK             NK</span></span>
<span><span class="co">## ...                        ...         ...            ...            ...</span></span>
<span><span class="co">## TTTGTTGGTTCAAGTC-1   cluster 5   cluster 4              B B intermediate</span></span>
<span><span class="co">## TTTGTTGGTTGCATGT-1   cluster 8   cluster 8          CD8 T      CD8 Naive</span></span>
<span><span class="co">## TTTGTTGGTTGCGGCT-1   cluster 2   cluster 3           Mono      CD14 Mono</span></span>
<span><span class="co">## TTTGTTGTCGAGTGAG-1   cluster 4   cluster 6        other T           MAIT</span></span>
<span><span class="co">## TTTGTTGTCGTTCAGA-1   cluster 4   cluster 6          CD4 T        CD4 TCM</span></span>
<span><span class="co">##                           RNA.Azimuth.L3 RNA.singleR.L1 RNA.singleR.L2</span></span>
<span><span class="co">##                              &lt;character&gt;    &lt;character&gt;    &lt;character&gt;</span></span>
<span><span class="co">## AAACCCAAGATTGTGA-1             CD14 Mono      Monocytes      Monocytes</span></span>
<span><span class="co">## AAACCCACATCGGTTA-1             CD14 Mono      Monocytes      Monocytes</span></span>
<span><span class="co">## AAACCCAGTACCGCGT-1             CD14 Mono      Monocytes      Monocytes</span></span>
<span><span class="co">## AAACCCAGTATCGAAA-1                  NK_1       NK cells       NK cells</span></span>
<span><span class="co">## AAACCCAGTCGTCATA-1                  NK_1       NK cells       NK cells</span></span>
<span><span class="co">## ...                                  ...            ...            ...</span></span>
<span><span class="co">## TTTGTTGGTTCAAGTC-1 B intermediate lambda       NK cells       NK cells</span></span>
<span><span class="co">## TTTGTTGGTTGCATGT-1             CD8 Naive   CD8+ T-cells   CD8+ T-cells</span></span>
<span><span class="co">## TTTGTTGGTTGCGGCT-1             CD14 Mono      Monocytes      Monocytes</span></span>
<span><span class="co">## TTTGTTGTCGAGTGAG-1                  MAIT   CD8+ T-cells       CD4+ Tem</span></span>
<span><span class="co">## TTTGTTGTCGTTCAGA-1             CD4 TCM_1   CD4+ T-cells       CD4+ Tcm</span></span>
<span><span class="co">##                    ADT.citesort    ADT.citesort.rename</span></span>
<span><span class="co">##                     &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">## AAACCCAAGATTGTGA-1     147_leaf CD45RO- CD16- monocyte</span></span>
<span><span class="co">## AAACCCACATCGGTTA-1     151_leaf CD45RA+ CD56+ CD16- ..</span></span>
<span><span class="co">## AAACCCAGTACCGCGT-1     125_leaf CD45RA+ CD45RO- CD16..</span></span>
<span><span class="co">## AAACCCAGTATCGAAA-1      63_leaf    CD16+ CD56+ NK cell</span></span>
<span><span class="co">## AAACCCAGTCGTCATA-1      63_leaf    CD16+ CD56+ NK cell</span></span>
<span><span class="co">## ...                         ...                    ...</span></span>
<span><span class="co">## TTTGTTGGTTCAAGTC-1      38_leaf CD16+ CD56+ TIGIT+ B..</span></span>
<span><span class="co">## TTTGTTGGTTGCATGT-1      23_leaf      naive CD8+ T cell</span></span>
<span><span class="co">## TTTGTTGGTTGCGGCT-1      67_leaf CD45RA+ CD45RO- CD16..</span></span>
<span><span class="co">## TTTGTTGTCGAGTGAG-1     169_leaf central memory CD4+ ..</span></span>
<span><span class="co">## TTTGTTGTCGTTCAGA-1     159_leaf central memory CD4+ ..</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="singlecellexperiment-object">
<code>SingleCellExperiment</code> object<a class="anchor" aria-label="anchor" href="#singlecellexperiment-object"></a>
</h3>
<p>Providing a <code>SingleCellExperiment</code> object, one also needs
to ensure that the data is formatted in a specific manner. First, data
should be normalized and stored as a <code>logcounts</code> assay within
the <code>SingleCellExperiment</code> object. For example, in the main
text, CITE-seq data was normalized using the R package <em>ADTnorm</em>.
Second, cell and marker names within the
<code>SingleCellExperiment</code> object should be defined (ensure these
are not set as <code>NULL</code>). Finally, different annotations should
be stored as columns of the <code>colData</code> within the object.
Likewise, the object can contain the principal components of the data
calculated using different decomposition techniques in
<code>reducedDims</code> (see <a href="https://doi.org/10.1038/s41592-019-0654-x" class="external-link">Amezquita et.
al.Â 2020</a> for further information). Notice that an example of a
<code>SingleCellExperiment</code> that can be used as input by
<code>tagtango</code> can be generated using:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span><span class="op">[[</span><span class="st">"ADT"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SingleCellExperiment </span></span>
<span><span class="co">## dim: 17 7472 </span></span>
<span><span class="co">## metadata(1): Samples</span></span>
<span><span class="co">## assays(1): logcounts</span></span>
<span><span class="co">## rownames(17): CD3 CD4 ... IgG1 IgG2b</span></span>
<span><span class="co">## rowData names(3): ID Symbol Type</span></span>
<span><span class="co">## colnames(7472): AAACCCAAGATTGTGA-1 AAACCCACATCGGTTA-1 ...</span></span>
<span><span class="co">##   TTTGTTGTCGAGTGAG-1 TTTGTTGTCGTTCAGA-1</span></span>
<span><span class="co">## colData names(20): Sample Barcode ... RNA.singleR.L2 ADT.citesort</span></span>
<span><span class="co">## reducedDimNames(1): UMAP</span></span>
<span><span class="co">## mainExpName: ADT</span></span>
<span><span class="co">## altExpNames(0):</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="data-frame-object">Data frame object<a class="anchor" aria-label="anchor" href="#data-frame-object"></a>
</h3>
<p>Providing a data frame object is the simplest way to run
<code>tagtango</code>. The application expects different annotations to
be stored as columns of the data frame. An example of a data frame that
can be used as input by <code>tagtango</code> can be generated again
with:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MultiAssayExperiment</span><span class="fu">::</span><span class="fu">colData</span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="supplementary-usage-scenario-comparing-spatial-transcriptomics-annotations">Supplementary usage scenario: comparing spatial transcriptomics
annotations<a class="anchor" aria-label="anchor" href="#supplementary-usage-scenario-comparing-spatial-transcriptomics-annotations"></a>
</h2>
<p>To highlight the versatility of <code>tagtango</code>, we tested our
software on a different data modality. We used an annotated spatial
transcriptomics dataset provided as part of the <a href="https://github.com/LieberInstitute/spatialLIBD" class="external-link"><em>spatialLIBD</em></a>
project (<a href="https://doi.org/10.1038/s41593-020-00787-0" class="external-link">Maynard
et. al., 2021;</a>, <a href="https://doi.org/10.1186/s12864-022-08601-w" class="external-link">Pardo et. al.,
2022</a>). The data was generated with 10x Genomics Visium platform and
contain human brain tissue samples from three healthy donors. In
particular, these are spatially adjacent replicates of human
dorsolateral prefrontal cortex tissue. This dataset is interesting
because it investigates the laminar structure of the brain cortex,
providing several manual annotations of the different layers for the
individual spots (considered as `ground truthâ€™) as well as several
cluster-based annotations.</p>
<p>Following the input data requirements outlined above, we prepared the
dataset with R to be studied with <code>tagtango</code>. First, we
followed <a href="https://doi.org/10.1186/s12864-022-08601-w" class="external-link">Pardo et.
al., (2022)</a>, and downloaded the spatial transcriptomics dataset.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load libraries</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LieberInstitute/spatialLIBD" class="external-link">spatialLIBD</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Download data</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/spatialLIBD/man/fetch_data.html" class="external-link">fetch_data</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"spe"</span><span class="op">)</span></span></code></pre></div>
<p>The data comes in the form of a <code>SpatialExperiment</code> R
object. We reshaped the data into a <code>SingleCellExperiment</code>
object using</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span> logcounts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">)</span>,</span>
<span>        colData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-methods.html" class="external-link">spatialCoords</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span> <span class="op">)</span>,</span>
<span>        rowData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span>
<span>     <span class="op">)</span></span>
<span>    </span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runUMAP.html" class="external-link">runUMAP</a></span><span class="op">(</span><span class="va">sce</span>, exprs_values <span class="op">=</span> <span class="st">"logcounts"</span><span class="op">)</span></span></code></pre></div>
<p>Notice that the spatial coordinates of the spatial transcriptomics
dataset were added as columns in the <code>colData</code> object (these
could also be added as part of the <code>reducedDims</code> object), and
that we calculated the UMAP decomposition using the R package
<em>scater</em> (<a href="https://doi.org/10.1093/bioinformatics/btw777" class="external-link">McCarthy et. al.,
2017</a>).</p>
<p>Finally, this object contains multiple samples for a spatial
transcriptomics dataset; therefore, the column names are not unique.
Likewise, row names (i.e.Â gene IDs) can also be more intuitive if we use
the gene names as opposed to ensembl names. To fix this, we used:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Row names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span></span>
<span></span>
<span><span class="co"># Fix col names (they were not unique)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Using <code>tagtango</code>, we then compared sets of annotations for
this dataset. A first interesting comparison is that between the
consensus manual annotations and one of the best performing unsupervised
clustering method used in (<a href="https://doi.org/10.1038/s41593-020-00787-0" class="external-link">Maynard et. al.,
2021</a>). In particular, we focused on the differences between â€˜ground
truthâ€™ annotations and annotations using highly variable genes from
<em>scran</em> (<a href="https://doi.org/10.1186/s13059-016-0947-7" class="external-link">Lun
et. al., 2016</a>), 50 PCs for dimension reduction, and spatial
coordinates as features (i.e.Â method <code>HVG\_PCA\_spatial</code> as
described in <a href="https://doi.org/10.1038/s41593-020-00787-0" class="external-link">Maynard et. al.,
2021</a>). Figure 1 explores the comparison between spots manually
labelled as White Matter (WM) but annotated as two different clusters in
the unsupervised approach. Notice that we used <code>tagtango</code> to
filter out the results, focusing on one sample (i.e.Â labelled as
<code>151675</code>) excluding non-tissue spots (i.e.Â classified as
<code>NA</code> in the manual annotations). This comparison identified
key genes driving the distinction between sub-populations within the
manually annotated WM region. These include known WM and L5 marker genes
such as MBP, which, coupled with the fact that spots in cluster 4 did
not intersect with spots manually labelled as L5 or L6, suggests
additional modularity within the WM layer. Similarly, a marker gene for
gray matter/neurons that was used to define the sample orientation by <a href="https://doi.org/10.1038/s41593-020-00787-0" class="external-link">Maynard et. al.,
2021</a>, SNAP25 is also identified as relevant. Finally, MOBP, a gene
identified as top 10th most variable markers across layers by the author
but that our comparison suggests this also show some level of modularity
within WM, was also selected as relevant.</p>
<div class="float">
<img src="figureS1.png" style="width:100.0%" alt="Figure 1: Overview of the comparison between layer annotations in the brain cortex dataset. Panel (a) displays a Sankey diagram comparing the manual and unsupervised annotations. The diagram was filtered using tagtango to only include cells from sample `151675â€™ and in-tissue spots. The coloured links in the diagram indicate the cell populations selected for White Matter (WM). Panel (b) displays a direct comparison of the normalized RNA expression for the two selected cell populations, including only markers selected as relevant. The colours of the bars match those of the selected links in panel (a). Panel (c) presents the spatial representation of all spots, where the colours of the points match those of the selected links in panel (a)."><div class="figcaption">Figure 1: Overview of the comparison between
layer annotations in the brain cortex dataset. Panel (a) displays a
Sankey diagram comparing the manual and unsupervised annotations. The
diagram was filtered using <code>tagtango</code> to only include cells
from sample `151675â€™ and in-tissue spots. The coloured links in the
diagram indicate the cell populations selected for White Matter (WM).
Panel (b) displays a direct comparison of the normalized RNA expression
for the two selected cell populations, including only markers selected
as relevant. The colours of the bars match those of the selected links
in panel (a). Panel (c) presents the spatial representation of all
spots, where the colours of the points match those of the selected links
in panel (a).</div>
</div>
<p>Notice that there are multiple types of comparisons that could be
performed with <code>tagtango</code>. For example, we could try to
understand differences between the way the different co-authors of the
study labelled the spots, annotations that are also provided with the R
package. Likewise, we could focus on comparing full layers, reproducing
the authors results and potentially identifying additional marker genes
separating the cortex layers.</p>
</div>
<div class="section level2">
<h2 id="supplementary-usage-scenario-understanding-batch-effects">Supplementary usage scenario: understanding batch effects<a class="anchor" aria-label="anchor" href="#supplementary-usage-scenario-understanding-batch-effects"></a>
</h2>
<p>An interesting use case for <code>tagtango</code> is the
identification of marker differences across batches. To illustrate this,
we used the same spatial dataset described in the previous section to
compare manually annotated brain layers to the donor IDs. Figure 2
explores the differences between WM in donors <code>Br8100</code> and
<code>Br5595</code>. In this case, we see strong differences in the
normalized expression of several interesting genes. For example, we see
large differences again for MBP, a marker gene that is central to the
manual annotations. Likewise, we see batch effects in other notable
genes such as CNP, identified by <a href="https://doi.org/10.1016/j.cell.2012.02.052" class="external-link">Zeng et. al., 2021</a>
as brain cortex cell-type marker genes conserved between Human and
Mouse. Finally, we see also differences for MOBP, highlighting this gene
as not only variable across layers but also across batches.</p>
<div class="float">
<img src="figureS2.png" style="width:100.0%" alt="Figure 2: Overview of the comparison between samples in the brain cortex dataset. Panel (a) displays a Sankey diagram comparing the manual annotations and donor ID. The diagram was filtered using tagtango to only include in-tissue spots. The coloured links in the diagram indicate the cell populations selected for deeper analysis. Panel (b) displays a direct comparison of the normalized RNA expression for the two selected cell populations, including only markers selected as relevant. The colours of the bars match those of the selected links in panel (a). Panel (c) presents the UMAP representation of the RNA expression for all spots, where the colours of the points match those of the selected links in panel (a)."><div class="figcaption">Figure 2: Overview of the comparison between
samples in the brain cortex dataset. Panel (a) displays a Sankey diagram
comparing the manual annotations and donor ID. The diagram was filtered
using <code>tagtango</code> to only include in-tissue spots. The
coloured links in the diagram indicate the cell populations selected for
deeper analysis. Panel (b) displays a direct comparison of the
normalized RNA expression for the two selected cell populations,
including only markers selected as relevant. The colours of the bars
match those of the selected links in panel (a). Panel (c) presents the
UMAP representation of the RNA expression for all spots, where the
colours of the points match those of the selected links in panel
(a).</div>
</div>
</div>
<div class="section level2">
<h2 id="supplementary-usage-scenario-comparing-single-cell-datasets">Supplementary usage scenario: comparing single-cell datasets<a class="anchor" aria-label="anchor" href="#supplementary-usage-scenario-comparing-single-cell-datasets"></a>
</h2>
<p>As the final usage case scenario, we showcased how
<code>tagtango</code> can be used to compare datasets. To do so, we used
two independent 10x datasets: PBMCs from a healthy donor obtained by 10x
Genomics from AllCells (<a href="https://www.10xgenomics.com/datasets/10-k-human-pbm-cs-5-v-2-0-chromium-x-2-standard-6-1-0" class="external-link">10x
Genomics, 2021</a>), and PBMCs form a diseased Acute Lymphoblastic
Leukemia donor obtained by 10x Genomics from Sanguine Biosciences (<a href="https://www.10xgenomics.com/datasets/10k_5p_Human_diseased_PBMC_ALL_Fresh" class="external-link">10x
Genomics, 2024</a>).</p>
<p>In order to analyse the datasets together with <code>tagtango</code>,
we first independently processed and annotated each of them. To do so,
we followed a three-step process: first, we downloaded the corresponding
h5 file from the 10x Genomics platform; we then processed each dataset
using the R packages <code>Seurat</code> (<a href="https://doi.org/10.1038/s41587-023-01767-y" class="external-link">Hao et. al.,
2023</a>), <code>SingleCellExperiment</code> (<a href="https://doi.org/10.1038/s41592-019-0654-x" class="external-link">Amezquita et. al.,
2020</a>) and <code>scater</code> (<a href="https://doi.org/10.1093/bioinformatics/btw777" class="external-link">McCarthy et. al.,
2017</a>); and we use the R packages <code>celldex</code> and
<code>SingleR</code> (<a href="https://doi.org/10.1038/s41590-018-0276-y" class="external-link">Liu et. al., 2019</a>)
to annotate the single-cell datasets, identifying the main cell types.
In R, the processing of each dataset is as follows:}</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load libraries</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/SingleR" class="external-link">SingleR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/celldex" class="external-link">celldex</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set path to the h5 file downloaded from www.10xgenomics.com</span></span>
<span><span class="va">path</span> <span class="op">&lt;-</span> <span class="st">"path_to_h5_file"</span></span>
<span></span>
<span><span class="co"># Load raw data</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/as.SingleCellExperiment.html" class="external-link">as.SingleCellExperiment</a></span><span class="op">(</span></span>
<span>       <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span></span>
<span>            <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/Read10X_h5.html" class="external-link">Read10X_h5</a></span><span class="op">(</span><span class="va">path</span><span class="op">)</span></span>
<span>       <span class="op">)</span></span>
<span> <span class="op">)</span></span>
<span></span>
<span><span class="co"># calculate the proportion of mitochondrial reads</span></span>
<span><span class="va">mt.genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"^MT-"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQCMetrics.html" class="external-link">addPerCellQC</a></span><span class="op">(</span><span class="va">sce</span>, subsets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Mito <span class="op">=</span> <span class="va">mt.genes</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># perform QC</span></span>
<span><span class="va">qc.lib</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/isOutlier.html" class="external-link">isOutlier</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">sum</span>, log<span class="op">=</span><span class="cn">TRUE</span>, type<span class="op">=</span><span class="st">"lower"</span><span class="op">)</span></span>
<span><span class="va">qc.nexprs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/isOutlier.html" class="external-link">isOutlier</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">detected</span>, log<span class="op">=</span><span class="cn">TRUE</span>, type<span class="op">=</span><span class="st">"lower"</span><span class="op">)</span></span>
<span><span class="va">qc.mito</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/isOutlier.html" class="external-link">isOutlier</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">subsets_Mito_percent</span>, type<span class="op">=</span><span class="st">"higher"</span><span class="op">)</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">discard</span> <span class="op">&lt;-</span> <span class="va">qc.lib</span> <span class="op">|</span> <span class="va">qc.nexprs</span> <span class="op">|</span> <span class="va">qc.mito</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span>,<span class="op">!</span><span class="va">sce</span><span class="op">$</span><span class="va">discard</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Normalize counts</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/librarySizeFactors.html" class="external-link">computeLibraryFactors</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html" class="external-link">logNormCounts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Annotate dataset with SingleR</span></span>
<span><span class="va">ref</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleR/man/datasets.html" class="external-link">BlueprintEncodeData</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">pred</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleR/man/SingleR.html" class="external-link">SingleR</a></span><span class="op">(</span>test<span class="op">=</span><span class="va">sce</span>, ref<span class="op">=</span><span class="va">ref</span>, labels<span class="op">=</span><span class="va">ref</span><span class="op">$</span><span class="va">label.main</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>, Main.labels <span class="op">=</span> <span class="va">pred</span><span class="op">$</span><span class="va">labels</span><span class="op">)</span></span></code></pre></div>
<p>This workflow provided us with two annotated
<code>SingleCellExperiment</code>, one for the healthy patient
(i.e.~<code>sce\_healthy</code>) and one for the cancer patient
(i.e.~<code>sce\_cancer</code>). The last step before analysing the
files with <code>tagtango</code> was to integrate these objects,
removing biologically irrelevant batch effects. To do so, we used
packages <code>scran</code> (<a href="https://doi.org/10.12688/f1000research.9501.2" class="external-link">Lun et. al.,
2016</a>) and <code>batchelor</code> (<a href="https://doi.org/10.1038/nbt.4091" class="external-link">Haghverdi et. al., 2018</a>) to
correct the log-expression values via linear regression:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load libraries</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">batchelor</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Find genes in common</span></span>
<span><span class="va">universe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sce_cancer_clean</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sce_healthy_clean</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Variance model</span></span>
<span><span class="va">dec_cancer_clean</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVarByPoisson.html" class="external-link">modelGeneVarByPoisson</a></span><span class="op">(</span><span class="va">sce_cancer</span><span class="op">)</span><span class="op">[</span><span class="va">universe</span>,<span class="op">]</span></span>
<span><span class="va">dec_healthy_clean</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVarByPoisson.html" class="external-link">modelGeneVarByPoisson</a></span><span class="op">(</span><span class="va">sce_healthy</span><span class="op">)</span><span class="op">[</span><span class="va">universe</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co"># Find HVGs</span></span>
<span><span class="va">combined.dec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/combineVar.html" class="external-link">combineVar</a></span><span class="op">(</span><span class="va">dec_cancer_clean</span>, <span class="va">dec_healthy_clean</span><span class="op">)</span></span>
<span><span class="va">chosen.hvgs</span> <span class="op">&lt;-</span> <span class="va">combined.dec</span><span class="op">$</span><span class="va">bio</span> <span class="op">&gt;</span> <span class="fl">0</span></span>
<span></span>
<span><span class="co"># Per-batch scaling normalization</span></span>
<span><span class="va">rescaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/batchelor/man/multiBatchNorm.html" class="external-link">multiBatchNorm</a></span><span class="op">(</span><span class="va">sce_cancer</span><span class="op">[</span><span class="va">universe</span>,<span class="op">]</span>, <span class="va">sce_healthy</span><span class="op">[</span><span class="va">universe</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="va">pbmc_cancer</span> <span class="op">&lt;-</span> <span class="va">rescaled</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">pbmc_healthy</span> <span class="op">&lt;-</span> <span class="va">rescaled</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Merge datasets and calculate TSNA</span></span>
<span><span class="va">rescaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/batchelor/man/rescaleBatches.html" class="external-link">rescaleBatches</a></span><span class="op">(</span><span class="va">pbmc_cancer</span>, <span class="va">pbmc_healthy</span><span class="op">)</span></span>
<span><span class="va">rescaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span><span class="va">rescaled</span>, subset_row<span class="op">=</span><span class="va">chosen.hvgs</span>,</span>
<span>                 exprs_values<span class="op">=</span><span class="st">"corrected"</span>,</span>
<span>                 BSPARAM<span class="op">=</span><span class="fu">BiocSingular</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/BiocSingularParam.html" class="external-link">RandomParam</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">rescaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runTSNE.html" class="external-link">runTSNE</a></span><span class="op">(</span><span class="va">rescaled</span>, dimred<span class="op">=</span><span class="st">"PCA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Copy annotations and batch information</span></span>
<span><span class="va">rescaled</span><span class="op">$</span><span class="va">batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">rescaled</span><span class="op">$</span><span class="va">batch</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cancer"</span>, <span class="st">"healthy"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">rescaled</span><span class="op">$</span><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">pbmc_cancer</span><span class="op">$</span><span class="va">Main.labels</span>, <span class="va">pbmc_healthy</span><span class="op">$</span><span class="va">Main.labels</span><span class="op">)</span></span></code></pre></div>
<p>Using <code>tagtango</code>, we then compared the batch information
and the annotations found with <code>singleR</code> (Figure 3). In
particular, we focused on the differences across cells annotated as CD4+
T-cells across the two datasets. Again, <code>tagtango</code>
highlighted differences in the normalized expression of several
interesting genes, all related to immune response in cancer. Notably,
JUNB has been identified in the past as a gatekeeper for a certain type
of lymphoid leukemia (<a href="https://doi.org/10.1038/sj.onc.1210285" class="external-link">Ott et. al., 2007</a>),
and LTB has been shown to promote the development of T-cell acute
lymphoblastic leukemia (<a href="https://doi.org/10.1111/bjh.13760" class="external-link">Fernandes et. al., 2015</a>).
Likewise, genes such as CD27 or IL-32 have been closely associated to
the clinical outcome and prognostic of acute lymphoblastic leukemia
patients (<a href="https://doi.org/10.1038/bcj.2017.55" class="external-link">Chen et. al.,
2017;</a> <a href="https://doi.org/10.1016/j.htct.2023.10.003" class="external-link">Abobakr
et. al., 2023;</a> <a href="https://doi.org/10.3389/fimmu.2022.837590" class="external-link">Shim et. al.,
2022</a>), and there are links between the expression of IL-7R and TCF7
with subsets of these patients (<a href="https://doi.org/10.1016/j.jbior.2018.09.012" class="external-link">Oliveira et. al.,
2019;</a> <a href="https://doi.org/10.1038/s41467-021-24442-9" class="external-link">Van
Thillo et. al., 2021</a>). Overall, <code>tagtango</code> was able to
quickly identify genes that are associated with lymphocyte development,
transcriptional regulation, and immune signaling pathways in acute
lymphoblastic leukemia patients. However, further work would be required
to validate these findings and ensure their biological relevance.</p>
<div class="float">
<img src="figureS3.png" style="width:100.0%" alt="Figure 3: Overview of the comparison between datasets using tagtango. Panel (a) displays a Sankey diagram comparing the batch information and main cell types. The coloured links in the diagram indicate the cell populations selected for deeper analysis. Panel (b) displays a direct comparison of the normalized and batch-corrected RNA expression for the two selected cell populations, including only markers selected as relevant. The colours of the bars match those of the selected links in panel (a). Panel (c) presents the TSNE representation of the RNA expression, where the colours of the points match those of the selected links in panel (a)."><div class="figcaption">Figure 3: Overview of the comparison between
datasets using <code>tagtango</code>. Panel (a) displays a Sankey
diagram comparing the batch information and main cell types. The
coloured links in the diagram indicate the cell populations selected for
deeper analysis. Panel (b) displays a direct comparison of the
normalized and batch-corrected RNA expression for the two selected cell
populations, including only markers selected as relevant. The colours of
the bars match those of the selected links in panel (a). Panel (c)
presents the TSNE representation of the RNA expression, where the
colours of the points match those of the selected links in panel
(a).</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Bernat Bramon Mora.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
